src/hotspot/share/classfile/symbolTable.cpp

```diff
@@ -923,14 +923,14 @@ void SymbolTable::print_histogram() {
   tty->print_cr("  Total removed            " SIZE_FORMAT_W(7), _symbols_removed);
   if (_symbols_counted > 0) {
     tty->print_cr("  Percent removed          %3.2f",
-          ((float)_symbols_removed / _symbols_counted) * 100);
+          ((float)_symbols_removed / (float)_symbols_counted) * 100);
   }
   tty->print_cr("  Reference counts         " SIZE_FORMAT_W(7), Symbol::_total_count);
   tty->print_cr("  Symbol arena used        " SIZE_FORMAT_W(7) "K", arena()->used() / K);
   tty->print_cr("  Symbol arena size        " SIZE_FORMAT_W(7) "K", arena()->size_in_bytes() / K);
   tty->print_cr("  Total symbol length      " SIZE_FORMAT_W(7), hi.total_length);
   tty->print_cr("  Maximum symbol length    " SIZE_FORMAT_W(7), hi.max_length);
-  tty->print_cr("  Average symbol length    %7.2f", ((float)hi.total_length / hi.total_count));
+  tty->print_cr("  Average symbol length    %7.2f", ((float)hi.total_length / (float)hi.total_count));
```

Comment:
Same here

Reply:
I think each value needs to be cast to get the correct result.

Reply:
Yes size_t values must be cast to float/double. Again I think double should be used here.

Reply:
Also changed to double.